packages/flutter/lib/src/rendering/table.dart

```diff
@@ -881,11 +881,11 @@ class RenderTable extends RenderBox {
         }
         assert(tableWidth >= targetWidth);
       }
-    } else // step 2 and 3 are mutually exclusive
+    } // step 2 and 3 are mutually exclusive
 
     // 3. if there were no flex columns, then grow the table to the
     //    minWidth.
-    if (tableWidth < minWidthConstraint) {
+    else if (tableWidth < minWidthConstraint) {
```

Comment:
I'd move the comments around a bit but I wouldn't leave the else if to start on its own line.

Reply:
We agree on the fact that `else if` should never be splitted.

I allowed this pattern for several reasons:

```dart
    var m;
    if (m) {
      m;
    }
    // my comment on this else branch
    else if (m) {
      m;
    }
    // my comment on this else branch
    else if (m) {
      m;
    }
```

looks better IMHO than end of line comments:

```dart
    var m;
    if (m) {
      m;
    } else if (m) { // my comment on this else branch that can not be multiline and should be short to not goes to far at right
      m;
    } else if (m) { // my comment on this else branch
      m;
    }
```

or

```dart
    var m;
    if (m) {
      m;
    } else if (m) {
      // my comment on this else branch here seems to reference the code that follows this comment and not the condition.
      m;
    } else if (m) {
      // my comment on this else branch
      m;
    }
```

Finally, `else if` may already start its own line without blocks:

```dart
    var m;
    if (m)
      m;
    else if (m)
      m;
```